lua/orgmode/org/links/handlers/_internal/custom_id.lua

```diff
@@ -0,0 +1,53 @@
+local utils = require('orgmode.utils')
+local Org = require('orgmode')
+local Internal = require('orgmode.org.hyperlinks.builtin.internal')
+
+---@class OrgLinkHandlerCustomId:OrgLinkHandlerInternal
+local CustomId = Internal:new()
+
+function CustomId:new(custom_id)
+  ---@class OrgLinkHandlerCustomId
+  local this = Internal:new()
+  this.custom_id = custom_id
+  setmetatable(this, self)
+  self.__index = self
+  return this
+end
+
+---@param input string
+function CustomId.parse(input)
+  return CustomId:new(input)
+end
+
+function CustomId:__tostring()
+  return string.format('#%s', self.custom_id)
+end
+
+function CustomId:follow()
+  local headlines = Org.files:get_current_file():find_headlines_with_property_matching('custom_id', self.custom_id)
```

Comment:
We should pass down `files` from the top through constructor instead of requiring `orgmode` directly.